### PR TITLE
[JENKINS-31679] Removed unnecessary System.err 

### PR DESCRIPTION
**Pull request by escoem**
[JENKINS-31679](https://issues.jenkins-ci.org/browse/JENKINS-31679)

@reviewbybees

### CODE REVIEW

File: src/main/java/jenkins/plugins/clangscanbuild/ClangScanBuildDescriptor.java [hunk 1]
```diff
@@ -105,8 +104,8 @@ public boolean configure( StaplerRequest req, JSONObject formData ) throws FormE
         return super.configure( req, formData );
     }
 
-    public boolean isApplicable( @SuppressWarnings("rawtypes") Class<? extends AbstractProject> jobType ){
-        return AbstractProject.class.isAssignableFrom( jobType );
+    @Override
+    public boolean isApplicable(Class<? extends AbstractProject> jobType) {
+    	return true;
```
Comment (recena) [inline src/main/java/jenkins/plugins/clangscanbuild/ClangScanBuildDescriptor.java]: Unrelated changes.

Comment (escoem) [inline src/main/java/jenkins/plugins/clangscanbuild/ClangScanBuildDescriptor.java]: Adds consistency to ClangScanBuildDescriptor isApplicable and ClangScanBuildPublisherDescriptor isApplicable methods after remove the System.err. All instances that receives these methods are AbstractProject.

Comment (recena) [inline src/main/java/jenkins/plugins/clangscanbuild/ClangScanBuildDescriptor.java]: I agree but the issue describes a different goal.

**Comment by tfennelly** [comment]
:bee: 

I agree with @recena wrt some of the changes appearing to be unrelated to the stated goal. That's not a biggie though imo ... just words ... we can see what you're doing :smile:

**Comment by amuniz** [comment]
:bee: (changes on `isApplicable` are equivalent to the code already existent and increase the readability, so +1).

**Comment by escoem** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
I would rather replace system errors to a logging on CONFIG/FINEST level since there is no test proofing that the plugin works for other AbstractProject types

**Comment by escoem** [comment]
Hi, I'm going to change logs to FINEST and recover AbstractProject.class.isAssignableFrom( jobType ) to as is suggested at PR-10

**Comment by stephenc** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
Re :bee: . Don't hesistate to ping

**Comment by escoem** [comment]
@reviewbybees done

Please @rodrigc, consider this PR ready to merge.

**Comment by rodrigc** [comment]
You had multiple different changes in a single commit.  Next time, try to separate different
changes into different commits.  It makes it easier to review.

**Comment by recena** [comment]
@rodrigc I agree with your comment but you should keep on the commit's author. I totally disagree with this practice.

File: src/main/java/jenkins/plugins/clangscanbuild/actions/ClangScanBuildAction.java [hunk 2]
```diff
@@ -77,10 +77,10 @@ public ClangScanBuildBugSummary loadBugSummary(){
 		        return null;
 		    }
 	    }catch( java.lang.InterruptedException ie ){
-			System.err.println( ie );
+	    	LOGGER.log(FINEST, "", ie);
 			return null;
 		}catch( IOException ioe ){
-			System.err.println( ioe );
+			LOGGER.log(FINEST, "", ioe);
```
Comment (xtofl) [inline src/main/java/jenkins/plugins/clangscanbuild/actions/ClangScanBuildAction.java]: I must say that, even when I would set the log level to INFO, I would like to be informed the reason why the summaryXML has not been loaded.  Unlike the `InterruptedException` (an user intended action), this case is a real error, so I would either keep the `System.err` or make it a `LOGGER.log(ERROR, "error reading bug summary XML", ioe);`

File: src/main/java/jenkins/plugins/clangscanbuild/actions/ClangScanBuildAction.java [hunk 3]
```diff
@@ -135,7 +135,7 @@ public void doBrowse( StaplerRequest req, StaplerResponse rsp ) throws IOExcepti
     	if( requestedPath == null ) rsp.sendError( 404 );
     
     	if( !APPROVED_REPORT_REQUEST_PATTERN.matcher( requestedPath ).matches() ){
-    		System.err.println( "Someone is requesting unapproved content: " + requestedPath );
+    		LOGGER.log(FINEST, "Someone is requesting unapproved content: %s", requestedPath);
```
Comment (xtofl) [inline src/main/java/jenkins/plugins/clangscanbuild/actions/ClangScanBuildAction.java]: If this deserves a 404 response, shouldn't it be a `WARNING`?

File: src/main/java/jenkins/plugins/clangscanbuild/actions/ClangScanBuildAction.java [hunk 4]
```diff
@@ -145,13 +145,13 @@ public void doBrowse( StaplerRequest req, StaplerResponse rsp ) throws IOExcepti
...
 	    	if( !requestedFile.exists() ){
-	    		System.err.println( "Unable to locate report: " + req.getRestOfPath() );
+	    		LOGGER.log(FINEST, "Unable to locate report: %s", req.getRestOfPath());
 	    		rsp.sendError( 404 );
 	    		return;
 	    	}
 	    	rsp.serveFile( req, requestedFile.toURI().toURL() );
     	}catch( Exception e ){
-    		System.err.println( "FAILED TO SERVE FILE: " + req.getRestOfPath() + " -> " + e.getLocalizedMessage() );
+    		LOGGER.log(FINEST, "FAILED TO SERVE FILE: %s -> %s", new Object[]{req.getRestOfPath(), e.getLocalizedMessage()});
```
Comment (xtofl) [inline src/main/java/jenkins/plugins/clangscanbuild/actions/ClangScanBuildAction.java]: An internal server error deserves a full blown `SEVERE` to me.
